atlasdb-backup/src/main/java/com/palantir/atlasdb/backup/BackupPersister.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.backup;
+
+import com.palantir.atlasdb.backup.api.BackupId;
+import com.palantir.atlasdb.backup.api.CompletedBackup;
+import com.palantir.atlasdb.internalschema.InternalSchemaMetadataState;
+import com.palantir.atlasdb.timelock.api.Namespace;
+import java.util.Optional;
+
+interface BackupPersister {
+    void storeSchemaMetadata(
+            BackupId backupId, Namespace namespace, InternalSchemaMetadataState internalSchemaMetadataState);
+
+    Optional<InternalSchemaMetadataState> getSchemaMetadata(BackupId backupId, Namespace namespace);
```

Comment:
If we can avoid having to get the schema metadata separately (hopefully we can do this), we can get rid of those two methods.

next PR will add an ExternalBackupPersister implementation, taking a `Supplier<Path>` or possibly `BiFunction<Namespace, BackupId, Path>`, to give us a folder into which to write our files (the identity and contents of which the internal backup product should not care about)

Reply:
This is still likely internal API, so you can have a type for everything that is persisted in the course of the backup. The internal product should ideally just provide the non-immutable parameters that are necessary to determine where to read/write from. However, this is ok for now -- anything that is not wire-format can be ironed out later